perfmetrics/scripts/load_tests/python/tasks/tf_data.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2023 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains tasks which reads data from files using tensorflow's tf.data api.
+
+Example:
+
+  lg_obj = load_generator.LoadGenerator(...)
+  task_obj = tf_data.TFDataRead(...)
+  observations = lg_obj.generate_load(task_obj)
+"""
+import os
+import logging
+import tensorflow as tf
+from load_generator import task
+
+# .tfrecord file contains multiple TFRecords in it. This defines the size of
+# that single TFRecord in bytes.
+SINGLE_RECORD_SIZE = 256 * 1024
+TF_DATA_READ = 'tf_data_read'
+
+
+class TFDataRead(task.LoadTestTask):
+  """Task class for reading tfrecord file using tensorflow's tf.data api.
+
+  Note: The same class can be used with tf's internal GCS client if format of
+  file path starts with gs:// and with GCSFuse if local GCSFuse mounted path
+  is passed.
+
+  For details on prefetch, num_parallel_calls & shard arguments, please refer
+  to tf.data public documentation.
```

Comment:
It would be good to add the public documentation link.

Reply:
done, added for tf.io.gfile